matrix_sdk/src/request_builder.rs

```diff
@@ -405,6 +406,92 @@ impl Into<register::Request> for RegistrationBuilder {
     }
 }
 
+/// Create a builder for making get_message_event requests.
+///
+/// # Examples
+/// ```
+/// # use std::convert::TryFrom;
+/// # use matrix_sdk::{Client, RoomSearchBuilder};
+/// # use matrix_sdk::api::r0::directory::get_public_rooms_filtered::{self, RoomNetwork, Filter};
+/// # use url::Url;
+/// # let homeserver = Url::parse("http://example.com").unwrap();
+/// # let mut rt = tokio::runtime::Runtime::new().unwrap();
+/// # rt.block_on(async {
+/// # let last_sync_token = "".to_string();
+/// let mut client = Client::new(homeserver).unwrap();
+///
+/// let generic_search_term = Some("matrix-rust-sdk".to_string());
+/// let mut builder = RoomSearchBuilder::new();
+/// builder
+///     .filter(Filter { generic_search_term, })
+///     .since(last_sync_token)
+///     .room_network(RoomNetwork::Matrix);
+///
+/// client.get_public_rooms_filtered(builder).await.is_err();
+/// # })
+/// ```
+#[derive(Clone, Debug, Default)]
+pub struct RoomSearchBuilder {
```

Comment:
Let us find a better name for this, this made me think of the server-side search for room events.

The spec almost nowhere mentions "room search" for this endpoint so people will not expect those keywords even if they do want to search for a room on the server. The method that accepts this struct also doesn't mention search in the name.

Besides that, we'll have trouble naming the filter builder for the server-side search if we do pick this name.


Something like PublicRoomsFilter or RoomListFilter sounds better to me, other suggestions are of course welcome.

Reply:
`RoomListFilterBuilder`

We still want it to be a "Builder" right?

Reply:
Sure, the other ones contain "Builder" in the name so lets keep it here as well.